app/views/incomeSources/manage/ManageObligations.scala.html

```diff
@@ -0,0 +1,204 @@
+                                <th scope="col" class="govuk-table__header"> </th>
+                            </tr>
+                        </thead>
+
+                        <tbody class="govuk-table__body">
+                        @sources.quarterlyObligationsDatesYearTwo.map { x =>
+                            <tr class="govuk-table__row">
+                                <td class="govuk-table__cell">@(x.inboundCorrespondenceFrom.toLongDate + " " + getMessage("to") + " " + x.inboundCorrespondenceTo.toLongDate)</td>
+                                <td class="govuk-table__cell">@(x.inboundCorrespondenceDue.toLongDate)</td>
+                            </tr>
+                        }
+                        </tbody>
+                    }
+                </table>
+            </div>
+        }
+
+        @if(sources.eopsObligationsDates.nonEmpty) {
+            <div class="box-simple" id="eops">
+                <div class="page_headers govuk-heading-m"> @getMessage("eops-heading") </div>
+                <p class="govuk-body"> @getMessage("eops-p1") </p>
+
+                <table class="govuk-table">
+                    <thead class="govuk-table__head">
+                        <tr class="govuk-table__row">
+                            <th scope="col" class="govuk-table__header">@getMessage("table.heading-1")</th>
+                            <th scope="col" class="govuk-table__header">@getMessage("table.heading-2")</th>
+                        </tr>
+                    </thead>
+
+                    <tbody class="govuk-table__body">
+                    @sources.eopsObligationsDates.map { x =>
+                        <tr class="govuk-table__row">
+                            <td class="govuk-table__cell">@(x.inboundCorrespondenceFrom.getYear + " " + getMessage("to") + " " + x.inboundCorrespondenceTo.getYear)</td>
+                            <td class="govuk-table__cell">@(x.inboundCorrespondenceDue.toLongDate)</td>
+                        </tr>
+                    }
+                    </tbody>
+                </table>
+            </div>
+        }
+
+
+        <div class="box-simple" id="finalDec">
+            <div class="page_headers govuk-heading-m"> @getMessage("final-declaration-heading") </div>
+            <p class="govuk-body"> @getMessage("final-declaration-text") </p>
+
+            <table class="govuk-table">
+                <thead class="govuk-table__head">
+                    <tr class="govuk-table__row">
+                        <th scope="col" class="govuk-table__header">@getMessage("table.heading-1")</th>
+                        <th scope="col" class="govuk-table__header">@getMessage("table.heading-2")</th>
+                    </tr>
+                </thead>
+
+                <tbody class="govuk-table__body">
+                @sources.finalDeclarationDates.map { x =>
```

Comment:
again, preferable to use .foreach, and you might move this to  re-usable code block is patterns is it repeating pattern